bundles/org.openhab.binding.gpstracker/src/main/java/org/openhab/binding/gpstracker/internal/handler/TrackerHandler.java

```diff
@@ -303,10 +303,11 @@ private void updateDistanceChannelFromMessage(LocationMessage message, Channel c
         Configuration currentConfig = c.getConfiguration();
         // convert into meters which is the unit of the threshold
         Double accuracyThreshold = convertToMeters(ConfigHelper.getAccuracyThreshold(currentConfig));
-        if (accuracyThreshold > message.getGpsAccuracy().doubleValue() || accuracyThreshold.intValue() == 0) {
+        Double accuracy = message.getGpsAccuracy() != null ? message.getGpsAccuracy().doubleValue() : accuracyThreshold;
```

Comment:
Could you please also annotate the `message` class with `@NonNullByDefault`? That way the compiler finds possible NPE violations.